resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -247,6 +241,18 @@ private[yarn] class ExecutorRunnable(
 
     System.getenv().asScala.filterKeys(_.startsWith("SPARK"))
       .foreach { case (k, v) => env(k) = v }
+
+    sparkConf.getExecutorEnv.foreach { case (key, value) =>
+      if (key == Environment.CLASSPATH.name()) {
```

Comment:
there are other variables that requires the append, like LD_LIBRARY_PATH.   It obviously can do bad things if its specified twice though but generally you shouldn't be doing that.  

Maybe I missed it in the description exactly how is AM env taking presendence?  Can you please describe the flow as to why moving this or changing it fixes the bug (how is executor getting am env now)?

Reply:
@tgravescs In existing code, in `prepareEnvironment`, "env" is populated only with `Environment.CLASSPATH`. Hence LD_LIBRARY_PATH does not apply to this specific change.

Reply:
Ah, I see sorry missed that. So I guess here we are just stomping on whatever is in the system env path now, vs before we were stomping on the executorEnv specified with the system env.